File: backend/src/main/java/com/tasklion/backend/model/TaskerModel.java
package com.tasklion.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class TaskerModel extends TasklionUserModel {

    private PersonalDetailModel personalDetail;
    private String aboutMe;
    private AddressModel address;
    private List<ServiceModel> services;

}

File: frontend/src/app/shared/models/tasker.model.ts
import {TasklionUserModel} from "./tasklion-user.model";
import {PersonalDetailModel} from "./personal-detail.model";
import {ServiceModel} from "./service.model";
import {AddressModel} from "./address.model";

export interface TaskerModel extends TasklionUserModel {

  personalDetail: PersonalDetailModel;
  aboutMe: string;
  address: AddressModel,
  services: ServiceModel[];

}
